Don't check stderr in configure tests as this causes them to fail when
unexpected warnings are emitted, such as deprecation of i386 in Xcode 10.

https://gitlab.freedesktop.org/cairo/cairo/-/issues/121

Updated version of this patch:

https://bugs.freedesktop.org/show_bug.cgi?id=30910#c6
--- configure.orig	2020-11-27 20:18:15.000000000 -0600
+++ configure	2021-01-24 22:11:20.000000000 -0600
@@ -18946,7 +18946,7 @@
 rm -f core conftest.err conftest.$ac_objext \
     conftest$ac_exeext conftest.$ac_ext
 
-	if test "x$cairo_cc_stderr" != "x"; then
+	if false; then
 		cairo_cc_flag=no
 	fi
 
@@ -18993,7 +18993,7 @@
 rm -f core conftest.err conftest.$ac_objext \
     conftest$ac_exeext conftest.$ac_ext
 
-	if test "x$cairo_cc_stderr" != "x"; then
+	if false; then
 		cairo_cc_flag=no
 	fi
 
@@ -19044,7 +19044,7 @@
 rm -f core conftest.err conftest.$ac_objext \
     conftest$ac_exeext conftest.$ac_ext
 
-	if test "x$cairo_cc_stderr" != "x"; then
+	if false; then
 		cairo_cc_flag=no
 	fi
 
@@ -19091,7 +19091,7 @@
 rm -f core conftest.err conftest.$ac_objext \
     conftest$ac_exeext conftest.$ac_ext
 
-	if test "x$cairo_cc_stderr" != "x"; then
+	if false; then
 		cairo_cc_flag=no
 	fi
 
@@ -19160,7 +19160,7 @@
 rm -f core conftest.err conftest.$ac_objext \
     conftest$ac_exeext conftest.$ac_ext
 
-	if test "x$cairo_cc_stderr" != "x"; then
+	if false; then
 		cairo_cc_flag=no
 	fi
 
@@ -19216,7 +19216,7 @@
 rm -f core conftest.err conftest.$ac_objext \
     conftest$ac_exeext conftest.$ac_ext
 
-	if test "x$cairo_cc_stderr" != "x"; then
+	if false; then
 		cairo_cc_flag=no
 	fi
 
@@ -20783,7 +20783,7 @@
 rm -f core conftest.err conftest.$ac_objext \
     conftest$ac_exeext conftest.$ac_ext
 
-	if test "x$cairo_cc_stderr" != "x"; then
+	if false; then
 		cairo_cc_flag=no
 	fi
 
@@ -20830,7 +20830,7 @@
 rm -f core conftest.err conftest.$ac_objext \
     conftest$ac_exeext conftest.$ac_ext
 
-	if test "x$cairo_cc_stderr" != "x"; then
+	if false; then
 		cairo_cc_flag=no
 	fi
 
@@ -34129,7 +34129,7 @@
 rm -f core conftest.err conftest.$ac_objext \
     conftest$ac_exeext conftest.$ac_ext
 
-	if test "x$cairo_cc_stderr" != "x"; then
+	if false; then
 		cairo_cc_flag=no
 	fi
 
@@ -34248,7 +34248,7 @@
 rm -f core conftest.err conftest.$ac_objext \
     conftest$ac_exeext conftest.$ac_ext
 
-	if test "x$cairo_cc_stderr" != "x"; then
+	if false; then
 		cairo_cc_flag=no
 	fi
 
@@ -34329,7 +34329,7 @@
 rm -f core conftest.err conftest.$ac_objext \
     conftest$ac_exeext conftest.$ac_ext
 
-	if test "x$cairo_cc_stderr" != "x"; then
+	if false; then
 		cairo_cc_flag=no
 	fi
 
